osf/management/commands/migrate_preprint_providers.py

```diff
@@ -0,0 +1,45 @@
+from django.core.management.base import BaseCommand
+
+from osf.models import Preprint, PreprintProvider
+
+
+"""
+A management command to migrate preperints from one provider to another.
+
+i.e. docker-compose run --rm web python3 manage.py migrate_preprint_providers --source_provider lawarxiv --destination_provider osf
+"""
+
+
+def migrate_preprint_providers(source_provider_guid, destination_provider_guid):
+    source_provider = PreprintProvider.load(source_provider_guid)
+    destination_provider = PreprintProvider.load(destination_provider_guid)
+    migration_count = 0
+
+    for preprint in Preprint.objects.filter(provider=source_provider):
```

Comment:
This is better as `Preprint.objects.filter(provider=source_provider).update(provider=destination_provider)`

Reply:
I actually had that originally but changed it because update() doesn't call the save() function of the model or send the post_save signal. It looked like preprints's save or post_save function may be necessary which is why I changed it to its current form.